Mesh_3/demo/Mesh_3/Volume_plane_thread.h

```diff
@@ -10,12 +10,13 @@
 #include <QThread>
 #include <vector>
 
+template<typename Word>
 struct Clamp_to_one_zero_range {
-  std::pair<float, float> min_max;
-  float operator()(const float& inVal) {
-    float inValNorm = inVal - min_max.first;
-    float aUpperNorm = min_max.second - min_max.first;
-    float bValNorm = inValNorm / aUpperNorm;
+  std::pair<Word, Word> min_max;
+  Word operator()(const Word& inVal) {
+    Word inValNorm = inVal - min_max.first;
+    Word aUpperNorm = min_max.second - min_max.first;
+    Word bValNorm = inValNorm / aUpperNorm;
```

Comment:
I am not sure that it the right way. In case `Word` is an integration type, that will not give you a nice mapping to the interval `[0., 1.]`.

I understand that `Clamp_to_one_zero_range` should have `Word` as template parameter, but the result of `operator()` should be a interval of the floating point type that the Qt widget requires. And use a `static_cast` at the right place to remove the warning.